tests/_runtime/test_runtime.py

```diff
@@ -1227,6 +1227,22 @@ async def test_temporaries_deleted(
         await k.run([ExecutionRequest(er.cell_id, "None")])
         assert f"_cell_{er.cell_id}_x" not in k.globals
 
+    async def test_has_run_id(
```

Comment:
I'm not sure why this test fails, `mocked_kernel.stream.messages` does contain run_id but`parse_raw` doesn't create a class with run_id.

Reply:
oh i accidentlly gave you a bug. it's trying to post-init again, 

```diff
    def __post_init__(self) -> None:
+      if self.run_id is not None:
+           return
        try:
            self.run_id = RUN_ID_CTX.get()
        except LookupError:
            # Be specific about the exception we're catching
            # The context variable hasn't been set yet
            # may want to explore why this is happening in the future
            self.run_id = None
        except Exception as e:
            LOGGER.error("Error getting run id: %s", str(e))
            self.run_id = None
```

Reply:
interesting..oh wait, yes I remember why this doesn't work. If you refresh the page, empty runs are created.

Reply:
okiee have a fix

Reply:
thanks for all the suggestions & backend work! I think dark theme was a good addition too